src/Cli/dotnet/Commands/Workload/WorkloadCommandBase.cs

```diff
@@ -107,20 +105,29 @@ public WorkloadCommandBase(
         TempDirectoryPath = !string.IsNullOrWhiteSpace(tempDirPath)
             ? tempDirPath
             : !string.IsNullOrWhiteSpace(parseResult.GetValue(WorkloadInstallCommandParser.TempDirOption))
-                ? parseResult.GetValue(WorkloadInstallCommandParser.TempDirOption)
+                ? parseResult.GetValue(WorkloadInstallCommandParser.TempDirOption)!
                 : PathUtilities.CreateTempSubdirectory();
 
         TempPackagesDirectory = new DirectoryPath(Path.Combine(TempDirectoryPath, "dotnet-sdk-advertising-temp"));
 
-        IsPackageDownloaderProvided = nugetPackageDownloader != null;
-        PackageDownloader = IsPackageDownloaderProvided ? nugetPackageDownloader : new NuGetPackageDownloader.NuGetPackageDownloader(TempPackagesDirectory,
-            filePermissionSetter: null,
-            new FirstPartyNuGetPackageSigningVerifier(),
-            nugetLogger,
-            Reporter,
-            restoreActionConfig: RestoreActionConfiguration,
-            verifySignatures: VerifySignatures,
-            shouldUsePackageSourceMapping: true);
+        if (nugetPackageDownloader is not null)
+        {
+            IsPackageDownloaderProvided = true;
+            PackageDownloader = nugetPackageDownloader;
+        }
+        else
+        {
+            IsPackageDownloaderProvided = false;
+            PackageDownloader = new NuGetPackageDownloader.NuGetPackageDownloader(
+                TempPackagesDirectory,
+                filePermissionSetter: null,
+                new FirstPartyNuGetPackageSigningVerifier(),
+                nugetLogger,
+                Reporter,
+                restoreActionConfig: RestoreActionConfiguration,
+                verifySignatures: VerifySignatures,
+                shouldUsePackageSourceMapping: true);
+        }
```

Comment:
This is just a simple expansion of the original code that is more NRT friendly. Could also change this to be a tuple return but this seemed more readable.